VegProcessor/species_hsi/blhwva.py

```diff
@@ -119,15 +134,20 @@ def blh_cover_mask(self, si_array: np.ndarray) -> np.ndarray:
         """To apply the BLH WVA, at least 40% BLH cover (Zone 3 to 5)
         has to be present. This applies a mask to each SI array where
         the %BLH cover is < 40%. These areas are given an SI = NaN.
+
+        Also ensures areas outside the hydro domain remain NaN.
         """
+        ## Apply hydro domain mask
+        si_array = np.where(np.isnan(self.hydro_domain_480), np.nan, si_array)
+
+        # Apply BLH cover mask
         if self.pct_blh_cover is not None:
-            blh_mask = (
```

Comment:
A few changes:
(1) `si_array` starts with the correct `hydro_domain` template
(2) the explicit nan check is removed, because in this instance nan propagation make it redundant. Because we only set a subset of the array to nan (`si_array[blh_mask] = np.nan`) the template nans are propagated.